### ISSUE TITLE
Authorizing Ember Data requests?

### ISSUE THREAD

**Issue by jagthedrummer** [paulcwatts/ember-cognito#102]
Hey there, thanks for the great add-on!

I've managed to get sign up/in working and now I'm trying to authorize Ember Data calls through API Gateway, and I can't quite get it to work.

Based on what I can gather from the docs for this project and ember-simple-auth I have this in `app/adapters/application.js`:

```javascript
import DS from 'ember-data';
import DataAdapterMixin from 'ember-simple-auth/mixins/data-adapter-mixin';
import { isPresent } from '@ember/utils';
import { inject as service } from '@ember/service';

export default DS.JSONAPIAdapter.extend(DataAdapterMixin, {
  host: 'https://xxx.execute-api.us-east-1.amazonaws.com',
  namespace: 'dev',
  session: service(),
  authorize(xhr) {
    let { access_token } = this.get('session.data.authenticated');
    if (isPresent(access_token)) {
      xhr.setRequestHeader('Authorization', access_token);
    }
  }
});
```

When I try to load data (for instance) I can see that the `Authorization` header is present on the request but I get a 403 `Forbidden` response.

I _think_ that I have Cognito, API Gateway and Lambda all configured properly. Using the [aws-api-gateway-cli-test](https://github.com/AnomalyInnovations/aws-api-gateway-cli-test) tool I can get a successful request by doing something like this:

```bash
$ apig-test \
  --username='jeremy+test@octolabs.com' \
  --password='Testing!123' \
  --user-pool-id='us-east-1_xxx' \
  --app-client-id='yyy' \
  --identity-pool-id='us-east-1:zzz' \
  --invoke-url='https://yyz.execute-api.us-east-1.amazonaws.com/dev' \
  --path-template='/notes' \
  --method='GET'
Authenticating with User Pool
Getting temporary credentials
Making API request
{ status: 200, statusText: 'OK', data: { data: [] } }
```

Am I missing something on the Ember side of things?

**Comment by jagthedrummer** [comment]
Oooooh, I just realized that I don't have anywhere in my Ember project to set the `identity-pool-id`. So I'm guessing that means that `ember-cognito` is only handling the User Pool side of things and not the Identity Pool. Any guidance on how I should handle that?

**Comment by jagthedrummer** [comment]
Ok, after getting farther into this I discovered that the code above _is_ working correctly once I finally set up my authorizer correctly.

Previously I'd been using the built-in `AWS_IAM` authorizer, which did not work.

<img width="864" alt="screen shot 2018-08-20 at 1 41 02 pm" src="https://user-images.githubusercontent.com/58702/44359977-02d33c80-a47f-11e8-9432-6098f2cddfb9.png">

I finally realized that I needed to create a custom authorizer of type "Cognito" and use that.

<img width="859" alt="screen shot 2018-08-20 at 1 41 24 pm" src="https://user-images.githubusercontent.com/58702/44360009-141c4900-a47f-11e8-9b25-8873dd7bfdb1.png">

<img width="467" alt="screen shot 2018-08-20 at 1 42 05 pm" src="https://user-images.githubusercontent.com/58702/44360021-1da5b100-a47f-11e8-9851-b61525f1d675.png">

Once I did that it's all working like a charm.
